contrib/fparser/fparser_ad.cc

```diff
@@ -333,13 +333,17 @@ typename ADImplementation<Value_t>::CodeTreeAD ADImplementation<Value_t>::D(cons
     case cFloor:
     case cCeil:
     case cTrunc:
+      throw RefuseToTakeCrazyDerivativeException;
```

Comment:
I don't know what cTrunc does, but why not allow returning 0 derivatives of floor() and ceil() ?  They've got the same "zero everywhere except at a jump" structure as the comparison operators do.